Add OpenTelemetry and Adafruit ServoKit Libraries to Rosdep

proffalken opened the pull request:
<!-- Thank you for contributing a change to the rosdistro. There are two primary types of submissions.
Please select the appropriate template from below: ROSDEP_RULE_TEMPLATE or DOC_INDEX_TEMPLATE

If you're making a new release with bloom please use bloom to create the pull request automatically (except for the naming review request which must be made manually).
If you've already run the release bloom has a `--pull-request-only` option you can use.-->

<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependencies to the rosdep database.

## Package name:

* adafruit-servokit-pip
* opentelemetry-api-pip
* opentelemetry-exporter-otlp-pip
* opentelemetry-sdk-pip

## Package Upstream Source:

* https://github.com/adafruit/Adafruit_CircuitPython_ServoKit
* https://github.com/open-telemetry/opentelemetry-python

## Purpose of using this:

These dependencies are proposed for the following reasons:
1. Adafruit ServoKit allows for control of standard hobby servos via the I2C PCA9685 driver boards, enabling small robot arms to be controlled via ROS easily on a Raspberry Pi or similar device
2. [Open Telemetry](https://www.opentelemetry.io) is rapidly becoming the defacto standard for monitoring software performance across the IT industry.  Adding these libraries enables us to visualise the interactions between ROS components in standard observability tooling such as Grafana, Honeycomb.io, or DataDog

Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/
  - N/A - Python Pip Packge
- Ubuntu: https://packages.ubuntu.com/
  - N/A - Python Pip Packge
- Fedora: https://packages.fedoraproject.org/
  - N/A - Python Pip Packge
- Arch: https://www.archlinux.org/packages/
  - N/A - Python Pip Packge
- Gentoo: https://packages.gentoo.org/
  - N/A - Python Pip Packge
- macOS: https://formulae.brew.sh/
  - N/A - Python Pip Packge
- Alpine: https://pkgs.alpinelinux.org/packages
  - N/A - Python Pip Packge
- NixOS/nixpkgs: https://search.nixos.org/packages
  - N/A - Python Pip Packge
- openSUSE: https://software.opensuse.org/package/
  - N/A - Python Pip Packge
- rhel: https://rhel.pkgs.org/
  - N/A - Python Pip Packge

proffalken committed:
Add  adafruit-circuitpython-servokit to rosdep

proffalken committed:
Add OTEL libraries

proffalken committed:
Add rosdeps for sqlalchemy and requests

mjcarroll commented:
Standard pip disclaimer: ROS packages that depend on pip keys cannot be released into a ROS distribution. They can only be depended on by from-source builds. Because of this, system packages are highly preferred to pip packages.

proffalken commented:
> Standard pip disclaimer: ROS packages that depend on pip keys cannot be released into a ROS distribution. They can only be depended on by from-source builds. Because of this, system packages are highly preferred to pip packages.

Thanks, unfortunately these packages are only available in pip format and I'm not keen to take on OS packaging for them.

Should I just continue to keep my own fork and use a custom rosdep format rather than asking for this to be merged?

mjcarroll commented:
> Should I just continue to keep my own fork and use a custom rosdep format rather than asking for this to be merged?

It's perfectly fine to merge them here, we just generally like to remind people that it's the case so they aren't surprised when they go to package something that depends on a pip package.

proffalken commented:
For completeness, this is the end result in Grafana when the otel libraries are included:

![image](https://github.com/ros/rosdistro/assets/262545/fc6a7c8a-8f5a-414e-b792-b2ad15a16abe)
![image](https://github.com/ros/rosdistro/assets/262545/dbe5ebf1-4250-4350-8fbb-637fbe3df129)
![image](https://github.com/ros/rosdistro/assets/262545/37d4f652-21e6-443d-a9ae-b54ea67fab4c)
![image](https://github.com/ros/rosdistro/assets/262545/061cfc0a-717a-4196-b5ad-874445700ec0)

proffalken committed:
Add adafruit-circuitpython-motor to rosdep for stepper/DC motors

sloretz commented on rosdep/python.yaml:
```diff
@@ -19,6 +19,16 @@ adafruit-mcp3008-pip:
   ubuntu:
     pip:
       packages: [Adafruit-MCP3008]
+adafruit-circuitpython-motor-pip:
```
Since these are pip only keys, please prefix them with `python3-`. That is `adafruit-circuitpython-motor-pip` should become `python3-adafruit-circuitpython-motor-pip`.

https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#pip-1

proffalken committed:
Add python3 prefix to adafruit pip libraries

clalancette commented on rosdep/python.yaml:
```diff
@@ -19,7 +19,17 @@ adafruit-mcp3008-pip:
   ubuntu:
     pip:
       packages: [Adafruit-MCP3008]
-adafruit-pca9685-pip:
```
While the advice that @sloretz gave you here was correct for new keys, we shouldn't change the old ones (even if they don't meet our current standards).  That's because it might break people who are already using this.  Please put this one back to:

```
adafruit-pca9685-pip
```

proffalken committed:
Revert breaking change in library name for pca9685 driver

nuclearsandwich committed:
Use OS-level wildcard for new pip keys.

rosdep has supported this for a good while now and the ROS core team
recently decided to use them for all newly added pip keys.
This hasn't been added to the review guidelines yet so it's perfectly
reasonable not to know this.

clalancette committed:
Alphabetize.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>
